python/ray/tune/tuner.py

```diff
@@ -0,0 +1,176 @@
+import traceback
+from typing import Any, Callable, Dict, List, Optional, Type, Union
+
+import ray
+
+from ray.tune import TuneError, ExperimentAnalysis
+from ray.tune.api_v2.convertible_to_trainable import ConvertibleToTrainable
+from ray.tune.callback import Callback
+from ray.tune.trainable import Trainable
+from ray.tune.impl.tuner_internal import TunerInternal
+from ray.tune.tuner_config import TuneAlgoConfig
+from ray.util import PublicAPI
+from ray.util.client.common import ClientActorHandle
+from ray.util.ml_utils.node import force_on_current_node
+
+
+########################################################################
+# The motivations behind the new Tuner API as part of Ray MLC effort are that
+# 1. Users can seamlessly transition from their training process to tuning process.
+# 2. Dataset and data preprocessing are now an integral part of training/tuning process.
+#   Bonus point: you can even tune dataset and preprocessing.
+
+# Some requirements on Tuner that may help you understand some implementation decisions:
+# 1. Tuner needs to work in ray client mode.
+# 2. Tuner needs to be restored in case of any failures.
+########################################################################
```

Comment:
Please clean up the documentation to be suitable for the public audience instead of internal facing.